src/v2/hooks/useAtomState.ts

```
@@ -0,0 +1,233 @@
+};
+
+/**
+ * Hook for managing atom state with history/undo functionality
+ *
+ * @param atom - The writable atom to manage
+ * @param maxHistorySize - Maximum number of history entries to keep
+ * @returns Object with value, setter, undo, redo, and history utilities
+ */
+// biome-ignore lint/suspicious/noExplicitAny: Jotai WritableAtom requires any types for flexibility
+export const useAtomStateWithHistory = <T>(
+  atom: WritableAtom<T, any, any>,
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)